### PR TITLE
[PYTHON API] move frontend bindings to pyopenvino + move MO to use new Python API

### PR DESCRIPTION
**Pull request by AlexeyLebedev1**
Details:
   -  changes from https://github.com/openvinotoolkit/openvino/pull/8870 are included
   -  move frontend bindings as part of pyopenvino
   -  Update MO to use new API

Tickets: 68545, 69176

### CODE REVIEW

File: runtime/bindings/python/src/pyopenvino/CMakeLists.txt [hunk 1]
```diff
@@ -57,7 +57,7 @@ else()
 endif()
 
 target_include_directories(${PROJECT_NAME} PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/..")
-target_link_libraries(${PROJECT_NAME} PRIVATE openvino::runtime openvino::frontend::manager)
+target_link_libraries(${PROJECT_NAME} PRIVATE openvino::runtime ${OFFLINE_TRANSFORMATIONS_LIB} openvino::frontend::manager)
```
Comment (ilya-lavrenov) [inline runtime/bindings/python/src/pyopenvino/CMakeLists.txt:60]: why is it needed? should be used only in MO / POT

Comment (AlexeyLebedev1) [reply runtime/bindings/python/src/pyopenvino/CMakeLists.txt:60]: I think it needs hear https://github.com/openvinotoolkit/openvino/blob/master/runtime/bindings/python/src/pyopenvino/core/offline_transformations.cpp#L9

File: runtime/bindings/python/src/openvino/helpers.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""openvino helper functions."""
+
+
+from openvino import Function
+from openvino.inference_engine import IENetwork
+
+# TODO: will be removed after full updating MO with new api
```
Comment (jiwaszki) [inline runtime/bindings/python/src/openvino/helpers.py:10]: Add ticket number from this task here.

**Review by jiwaszki — CHANGES_REQUESTED** [review]
Fix code style as flake in CI detect some errors.

File: runtime/bindings/python/src/openvino/helpers.py (hunk 2 above)
Comment (akuporos) [reply runtime/bindings/python/src/openvino/helpers.py:10]: @AlexeyLebedev1 you may add this ticket 69196 -- I think it'll be one scope of work there

**Review by jiwaszki — COMMENTED** [review]
From bindings' perspective it looks good to me.

**Review by jiwaszki — APPROVED** [review]

**Review by GlebKazantaev — COMMENTED** [review]
@AlexeyLebedev1 what is the current status for this PR?
These changes are required by MO transition to the new API.

File: model-optimizer/mo/moc_frontend/serialize.py [hunk 3]
```diff
@@ -7,8 +7,8 @@
 from mo.back.ie_ir_ver_2.emitter import append_ir_info
 from mo.utils.cli_parser import get_meta_info, parse_transform
 
-from ngraph import Function         # pylint: disable=no-name-in-module,import-error
-from ngraph import function_to_cnn  # pylint: disable=no-name-in-module,import-error
+from openvino import Function         # pylint: disable=no-name-in-module,import-error
+from openvino import function_to_cnn  # pylint: disable=no-name-in-module,import-error
```
Comment (ilya-lavrenov) [inline model-optimizer/mo/moc_frontend/serialize.py:11]: I suppose `openvino` should not contain `CNNNetwork`

File: runtime/bindings/python/src/openvino/helpers.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+from openvino import Function
+from openvino.inference_engine import IENetwork
+
+
+# TODO: will be removed after full updating MO with new api (cvs-69196)
+def function_to_cnn(ng_function: Function) -> IENetwork:
+    """Get Inference Engine CNN network from openvino function."""
+    capsule = Function.to_capsule(ng_function)
+    return IENetwork(capsule)
```
Comment (nosovmik) [inline runtime/bindings/python/src/openvino/helpers.py:14]: Does it work well? I'm getting error "PyCapsule_GetPointer called with incorrect name" at ie_api.pyx when trying to convert serialize PaddlePaddle model...

Comment (akuporos) [reply runtime/bindings/python/src/openvino/helpers.py]: removed

File: model-optimizer/mo/moc_frontend/serialize.py (hunk 3 above)
Comment (akuporos) [reply model-optimizer/mo/moc_frontend/serialize.py]: removed

**Comment by akuporos** [comment]
> @AlexeyLebedev1 what is the current status for this PR? These changes are required by MO transition to the new API.

@GlebKazantaev I updated MO to use new API. 
Please, review.
Offline_transformation_pybind will be replaced with the new name `offline_transformation` in a different PR.

**Comment by azhogov** [comment]
CI passed: https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/506

**Pull request merged**